src/ImageSharp/Image/ImageBase{TColor}.cs

```diff
@@ -67,15 +84,24 @@ protected ImageBase(ImageBase<TColor> other)
             this.Height = other.Height;
             this.CopyProperties(other);
 
-            // Copy the pixels. Unsafe.CopyBlock gives us a nice speed boost here.
-            this.pixelBuffer = new TColor[this.Width * this.Height];
+            // Rent then copy the pixels. Unsafe.CopyBlock gives us a nice speed boost here.
+            this.RentPixels();
             using (PixelAccessor<TColor> sourcePixels = other.Lock())
             using (PixelAccessor<TColor> target = this.Lock())
             {
+                // Check we can do this without crashing
                 sourcePixels.CopyTo(target);
             }
         }
 
+        /// <summary>
+        /// Finalizes an instance of the <see cref="ImageBase{TColor}"/> class.
+        /// </summary>
+        ~ImageBase()
```

Comment:
Don't have finalizer, else if the user doesn't call dispose the allocated memory will hang around in the finalizer queue. Better to allow it to be picked up by GC early. 

Ideally the user would call Dispose, but if they don't without the finalizer the allocations will be the same as vanilla new; with the finalizer the allocations will increase.